src/server/scripts/Outland/TempestKeep/Mechanar/boss_nethermancer_sepethrea.cpp

```diff
@@ -15,77 +15,92 @@
+    SPELL_QUELL_RAGING_FLAMES      = 35277,
     SPELL_ARCANE_BLAST             = 35314,
     SPELL_DRAGONS_BREATH           = 35250,
-    SPELL_KNOCKBACK                = 37317,
-    SPELL_SOLARBURN                = 35267,
-    H_SPELL_SUMMON_RAGIN_FLAMES    = 39084, // Not scripted
-    SPELL_INFERNO                  = 35268, // Not scripted
-    H_SPELL_INFERNO                = 39346, // Not scripted
-    SPELL_FIRE_TAIL                = 35278  // Not scripted
+
+    // Raging Flames
+    SPELL_RAGING_FLAMES_DUMMY      = 35274, // NYI, no clue what it can do
+    SPELL_RAGING_FLAMES_AREA_AURA  = 35281,
+    SPELL_INVIS_STEALTH_DETECTION  = 18950, // For some reason every time it's used by summoned creatures, it's applied after random delay. Ignored
+    SPELL_INFERNO                  = 35268,
+    SPELL_INFERNO_DAMAGE           = 35283
 };
 
 enum Events
 {
-    EVENT_FROST_ATTACK             = 1,
-    EVENT_ARCANE_BLAST             = 2,
-    EVENT_DRAGONS_BREATH           = 3,
-    EVENT_KNOCKBACK                = 4,
-    EVENT_SOLARBURN                = 5
+    EVENT_ARCANE_BLAST             = 1,
+    EVENT_DRAGONS_BREATH
 };
 
 struct boss_nethermancer_sepethrea : public BossAI
 {
     boss_nethermancer_sepethrea(Creature* creature) : BossAI(creature, DATA_NETHERMANCER_SEPRETHREA) { }
 
+    void Reset() override
+    {
+        DoCastSelf(SPELL_FROST_ATTACK);
+        me->SetCombatPulseDelay(0);
+        events.Reset();
+        instance->SetBossState(DATA_NETHERMANCER_SEPRETHREA, NOT_STARTED);
+    }
+
     void JustEngagedWith(Unit* who) override
     {
         BossAI::JustEngagedWith(who);
-        events.ScheduleEvent(EVENT_FROST_ATTACK, 7s, 10s);
-        events.ScheduleEvent(EVENT_ARCANE_BLAST, 12s, 18s);
-        events.ScheduleEvent(EVENT_DRAGONS_BREATH, 18s, 22s);
-        events.ScheduleEvent(EVENT_KNOCKBACK, 22s, 28s);
-        events.ScheduleEvent(EVENT_SOLARBURN, 30s);
+        events.ScheduleEvent(EVENT_ARCANE_BLAST, 15s, 25s);
+        events.ScheduleEvent(EVENT_DRAGONS_BREATH, 20s, 30s);
         Talk(SAY_AGGRO);
-        DoCast(who, SPELL_SUMMON_RAGIN_FLAMES);
-        Talk(SAY_SUMMON);
+        DoCastSelf(SPELL_SUMMON_RAGING_FLAMES);
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (victim->GetTypeId() == TYPEID_PLAYER)
+            Talk(SAY_SLAY);
+    }
+
+    void EnterEvadeMode(EvadeReason why) override
+    {
+        // Fails because target is in evade mode (yes, she kills them on evade too)
+        // DoCastSelf(SPELL_QUELL_RAGING_FLAMES, true);
+        summons.DespawnAll();
+        ScriptedAI::EnterEvadeMode(why);
     }
 
-    void KilledUnit(Unit* /*victim*/) override
+    void JustReachedHome() override
     {
-        Talk(SAY_SLAY);
+        _JustReachedHome();
+        DoCastSelf(SPELL_FROST_ATTACK);
     }
 
     void JustDied(Unit* /*killer*/) override
     {
-        _JustDied();
+        events.Reset();
+        instance->SetBossState(DATA_NETHERMANCER_SEPRETHREA, DONE);
```

Comment:
why did you remove _JustDied()  here?  This will lead to summons not getting despawned when the encounter ends so you will have orphaned spawns here.

Reply:
They're killed by SPELL_QUELL_RAGING_FLAMES

Reply:
Then don't store them at all.  

`void JustSummoned(Creature* /*summon*/) override { }`

This way you can cast the spell easily on evade and death and use the default ai hooks.

Reply:
I still despawn them on evade with summons.DespawnAll();

```cpp
    void EnterEvadeMode(EvadeReason why) override
    {
        // Fails because target is in evade mode (yes, she kills them on evade too)
        // DoCastSelf(SPELL_QUELL_RAGING_FLAMES, true);
        summons.DespawnAll();
```

Reply:
And why did you comment the spell then instead of using it?

Reply:
If it will be still safe to call both kill and despawn when kill will be fixed, I can uncomment it

Reply:
That's probably dumb logic, I'll uncomment it later